azuread/resource_group_test.go

```diff
@@ -116,10 +231,112 @@ func testCheckAzureADGroupDestroy(s *terraform.State) error {
 	return nil
 }
 
+func assertResourceWithMemberCount(id string, memberCount string) resource.TestCheckFunc {
+	resourceName := "azuread_group.test"
+
+	return resource.ComposeTestCheckFunc(
+		testCheckAzureADGroupExists(resourceName),
+		resource.TestCheckResourceAttr(resourceName, "name", fmt.Sprintf("acctest%s", id)),
+		resource.TestCheckResourceAttrSet(resourceName, "object_id"),
+		resource.TestCheckResourceAttr(resourceName, "members.#", memberCount),
+	)
+}
+
 func testAccAzureADGroup(id string) string {
 	return fmt.Sprintf(`
 resource "azuread_group" "test" {
   name = "acctest%s"
 }
 `, id)
 }
+
+func testAccAzureADGroupWithDiverseMembers(id string) string {
+	var sb strings.Builder
+
+	sb.WriteString(servicePrincipal(id))
+	sb.WriteString(group(id))
+	sb.WriteString(user(id))
+	sb.WriteString(groupWithMembers(id, fmt.Sprintf("azuread_user.acctest_user_%[1]s.object_id, azuread_group.test_g_%[1]s.object_id, azuread_service_principal.test_sp_%[1]s.object_id", id)))
+
+	return sb.String()
+}
+
+func testAccAzureADGroupWithMembers(id string, members []string) string {
+	var sb strings.Builder
+
+	sb.WriteString(`
+data "azuread_domains" "tenant_domain" {
+	only_initial = true
+}`)
+
+	for _, member := range members {
+		fmt.Fprintf(&sb, `
+resource "azuread_user" "acctest_user_%[1]s" {
+	user_principal_name   = "acctest.%[1]s@${data.azuread_domains.tenant_domain.domains.0.domain_name}"
+	display_name          = "acctest-%[1]s"
+	password              = "%[1]s"
+}
+`, member)
+	}
+
+	fmt.Fprintf(&sb, `
+resource "azuread_group" "test" {
+  name   = "acctest%s"
+  members = [ %s ]
+}
+`, id, strings.Join(formatMembersAsUser(members), ", "))
+
+	return sb.String()
+}
+
+func formatMembersAsUser(members []string) []string {
+	vsm := make([]string, len(members))
+	for i, v := range members {
+		vsm[i] = "azuread_user.acctest_user_" + v + ".object_id"
+	}
+	return vsm
+}
+
+func servicePrincipal(id string) string {
+	return fmt.Sprintf(`
+resource "azuread_application" "test_app_%[1]s" {
+  name = "app%[1]s"
+}
+
+resource "azuread_service_principal" "test_sp_%[1]s" {
+  application_id = azuread_application.test_app_%[1]s.application_id
+}
+
+`, id)
+}
+
+func group(id string) string {
```

Comment:
You could probably reuse the test config for a single group?

Reply:
If using `testAccAzureADGroup(id)` instead of `group(id)`, duplicate resources named `azuread_group.test` will exist, and therefore fail.

Reply:
you could change it to use `azuread_group.test%s` and pass in an additional argument?